pkg/utils/memory_resources.go

```diff
@@ -24,124 +24,90 @@ import (
 	"k8s.io/klog/v2"
 )
 
 var (
-	sysBusNodeDevices = "/sys/bus/node/devices"
+	sysBusNodeBasepath = "/sys/bus/node/devices"
 )
 
-const (
-	HugepageSize2Mi = 2048
-	HugepageSize1Gi = 1048576
-)
+// NumaMemoryResources contains information of the memory resources per NUMA
+// nodes of the system.
+type NumaMemoryResources map[int]MemoryResourceInfo
 
-type MemoryResourcePerNUMA map[int]map[v1.ResourceName]int64
+// MemoryResourceInfo holds information of memory resources per resource type.
+type MemoryResourceInfo map[v1.ResourceName]int64
 
-// GetMemoryResourceCounters returns total amount of memory and hugepages under NUMA nodes
-func GetMemoryResourceCounters() (MemoryResourcePerNUMA, error) {
-	numaNodeIDs, err := getNUMANodesIDs()
+// GetNumaMemoryResources returns total amount of memory and hugepages under NUMA nodes
+func GetNumaMemoryResources() (NumaMemoryResources, error) {
+	nodes, err := ioutil.ReadDir(sysBusNodeBasepath)
 	if err != nil {
 		return nil, err
 	}
 
-	memoryResources := make(MemoryResourcePerNUMA)
-	for _, numaNodeID := range numaNodeIDs {
-		memoryResources[numaNodeID] = map[v1.ResourceName]int64{}
-		numaNodeFolderName := fmt.Sprintf("node%d", numaNodeID)
+	memoryResources := make(NumaMemoryResources, len(nodes))
+	for _, n := range nodes {
+		numaNode := n.Name()
+		nodeID, err := strconv.Atoi(numaNode[4:])
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse NUMA node ID of %w", numaNode)
+		}
+
+		info := make(MemoryResourceInfo)
 
-		// get the NUMA node total memory
-		memInfoFilePath := filepath.Join(sysBusNodeDevices, numaNodeFolderName, "meminfo")
-		nodeTotalMemory, err := readTotalMemoryFromMeminfo(memInfoFilePath)
+		// Get total memory
+		nodeTotalMemory, err := readTotalMemoryFromMeminfo(filepath.Join(sysBusNodeBasepath, numaNode, "meminfo"))
 		if err != nil {
 			return nil, err
 		}
-		memoryResources[numaNodeID][v1.ResourceMemory] = nodeTotalMemory
+		info[v1.ResourceMemory] = nodeTotalMemory
 
-		// get the NUMA node hugepages
-		hugepagesFolderPath := filepath.Join(sysBusNodeDevices, numaNodeFolderName, "hugepages")
-		hugepageSizes, err := getHugepagesSizes(hugepagesFolderPath)
+		// Get hugepages
+		hugepageCounts, err := getHugepagesCounts(filepath.Join(sysBusNodeBasepath, numaNode, "hugepages"))
 		if err != nil {
 			return nil, err
 		}
-
-		for _, hugepageSize := range hugepageSizes {
-			hugepagesSizeFolderName := fmt.Sprintf("hugepages-%dkB", hugepageSize)
-			hugepagesTotalFile := filepath.Join(
-				hugepagesFolderPath,
-				hugepagesSizeFolderName,
-				"nr_hugepages",
-			)
-			nodeHugepagesTotal, err := readIntFromFile(hugepagesTotalFile)
-			if err != nil {
-				return nil, err
-			}
-
-			hugepagesResource := hugepageResourceNameFromSize(hugepageSize)
-			// save the total amount of memory allocated for hugepages in bytes
-			memoryResources[numaNodeID][hugepagesResource] = int64(nodeHugepagesTotal * hugepageSize * 1024)
+		if nr, ok := hugepageCounts["2048kB"]; ok {
```

Comment:
I prefer to have more generic logic because the size of hugepages can be different from arch to arch, for example, ARM is supporting 16Mi hugepages.

Reply:
Woops, good point 🙄 I think I was a bit too hasty. Changed this, now